internal/app/server/open_saves.go

```diff
@@ -1012,6 +1012,33 @@ func (s *openSavesServer) cacheRecord(ctx context.Context, r *record.Record, hin
 	return err
 }
 
+func (s *openSavesServer) getStoreAndCache(ctx context.Context, storeKey string) (*store.Store, error) {
+	st := new(store.Store)
+	if err := s.cacheStore.Get(ctx, store.CacheKey(storeKey), st); err == nil {
+		log.Debug("store cache hit")
+		return st, nil
+	}
+	log.Debug("store cache miss")
+
+	str, err := s.metaDB.GetStore(ctx, storeKey)
+	if err != nil {
+		log.Warnf("GetStore failed for store (%s): %v",
+			storeKey, err)
+		return nil, status.Convert(err).Err()
+	}
+	s.storeCache(ctx, str)
```

Comment:
Can you please catch the error and log a warning here?

Since we never update a store, failing to add the store to the cache does not have any side effect. This is different for our records, where we end up with stale objects in the cache. We have another issue that will take care of retrying updating the cache for the record.